src/printers/flow.js

```diff
@@ -19,11 +19,67 @@ export default {
     ? Tokens.string('true')
     : Tokens.string('false'),
 
+  ExistentialTypeParam: () => [
+    Tokens.string('*'),
+  ],
+
+  FunctionTypeAnnotation: ({node, print, path}) => [
+    print(node.typeParameters),
+    Tokens.string('('),
+    (node.params.length || node.rest) && [
+      Tokens.scopeOpen('object'),
+      Tokens.scopeEmptyOrBreak(),
+      map(node.params, (param, i, arr) => [
+        i > 0 && Tokens.scopeSpaceOrBreak(),
+        print(param),
+        arr.length - 1 !== i && !node.rest
+          ? Tokens.comma()
+          : Tokens.scopeEmptyOrComma(),
+      ]),
+      node.rest && [
+        node.params.length && [
+          Tokens.comma(),
+          Tokens.scopeSpaceOrBreak(),
+        ],
+        Tokens.string('...'),
+        print(node.rest),
+        Tokens.scopeEmptyOrComma(),
+      ],
+      Tokens.scopeEmptyOrBreak(),
+      Tokens.scopeClose(),
+    ],
+    Tokens.string(')'),
+    (
+      path[path.length - 2].type === 'ObjectTypeProperty' ||
```

Comment:
I think i generally prefer the approach of the parent switching but for this case it would mean the parent will need to render this full node which can make it hard to find the logic and a little weird to find the logic for a given node. At least with the children switching its all self contained in a single fn.